lib/rules/comment-pattern/index.js

```diff
@@ -0,0 +1,47 @@
+// @ts-nocheck
+
+'use strict';
+
+const _ = require('lodash');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+
+const ruleName = 'comment-pattern';
+
+const messages = ruleMessages(ruleName, {
+	expected: 'Expected comment to match specified pattern',
+});
+
+function rule(pattern) {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: pattern,
+			possible: [_.isRegExp],
+		});
+
+		if (!validOptions) {
+			return;
+		}
```

Comment:
It needs to support string as I described in the other comment. You can use this code: https://github.com/stylelint/stylelint/blob/924646681a355b327ccfeb6f00be4e1636333911/lib/rules/custom-property-pattern/index.js#L19-L28